### PR TITLE
Feat: add new toggle UI component

### LINKED ISSUE #2085: Create new Toggle UI component
Create a Toggle Ui component following the new UI components standards

### PR DESCRIPTION
**Pull request by davidejensen**
## What does this PR change?

Fix #2085
Adds a new base UI component for toggles

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by Suduck — APPROVED** [review]

**Review by sandrade-dcl — APPROVED** [review]
Good work! The only thing you missed is to add the documentation of the new `Toggle_Capsule` UI Component in the notion documentation: https://www.notion.so/decentraland/UI-Components-Reference-f467a8d05d594718a2c12809dcfaf132.

**Review by sandrade-dcl — CHANGES_REQUESTED** [review]
Oh, I forgot it... you have to implement the tests for the new UI component.

**Review by AjimenezDCL — APPROVED** [review]
Approving with small changes

File: unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs
Comment (sandrade-dcl) [inline unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs]: This `Destroy` is not needed because it is already added into the `Dispose` in the `BaseComponentView` class.

File: unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs
Comment (sandrade-dcl) [inline unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs]:
```suggestion
        toggleComponent.onToggleChange.AddListener((isToggleOn)=>isOn = isToggleOn);
```

File: unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs
Comment (davidejensen) [reply unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs:24]: True, fixed

**Review by sandrade-dcl — CHANGES_REQUESTED** [review]
You should add tests for the next functions too:
- `SetTextActive()`
- `SetInteractable()`
- `IsInteractable()`

File: unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs
Comment (sandrade-dcl) [reply unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs]: @AjimenezDCL is this naming something standarize by us? I'm using the "_____Tests" way in all our UI Components tests.

Comment (AjimenezDCL) [reply unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs]: Oh, we kinda agreed to it last year or so but it has not been written in stone anywhere, I leave it up to you Davide, keep the UIComponents convention or the project-wide one, it's fine anyway.

**Review by sandrade-dcl — APPROVED** [review]
LGTM! Please, remember to add the documentation of the new Toggle component in Notion 👍

File: unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs
Comment (davidejensen) [reply unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs]: I firstly wrote it as ___Tests because it was the same as the button component one, but at the same time the world Tests is redundant and doesn't give that much info, I kinda prefer the "Should" because it forces the tests to be fluent.

Update: just talked with Santi, will leave the ___Tests to fit the rest of UIComponents

Comment (sandrade-dcl) [reply unity-renderer/Assets/UIComponents/Tests/ToggleComponentViewTests.cs:4]: Yeah, for now I prefer to keep these tests with the same namings convention as the rest of them are following the other way. But we can rename all of them in another future PR in order to standarize it 👍

**Pull request merged**
